### PR TITLE
Add Support to Show Warning for Past Events in Contribution Flow

### PR DESCRIPTION
**Pull request by SudharakaP**
Related to https://github.com/opencollective/opencollective/issues/3982

Related Front-end PR: https://github.com/opencollective/opencollective-frontend/pull/6797

### CODE REVIEW

File: server/graphql/v2/object/Event.js [hunk 1]
```diff
@@ -51,6 +52,16 @@ export const Event = new GraphQLObjectType({
...
         },
       },
+      endsAt: {
+        description: 'The Event end date and time',
+        type: GraphQLDateTime,
+        async resolve(event) {
+          if (!event.endsAt) {
+            return null;
+          }
+          return new Date(event.endsAt);
```
Comment (znarf) [inline server/graphql/v2/object/Event.js:62]: What justifies a such complicated resolver? Is a custom resolver actually needed? See `createdAt` on most objects.

File: server/graphql/v2/object/Event.js [hunk 2]
```diff
@@ -51,6 +52,16 @@ export const Event = new GraphQLObjectType({
           }
         },
       },
+      endsAt: {
```
Comment (znarf) [inline server/graphql/v2/object/Event.js:55]: Why not introduce `startsAt`? `endsAt` without `startsAt` looks a bit like a missed opportunity.

**Review by znarf — COMMENTED** [review]
Changes requested.

File: server/graphql/v2/object/Event.js (hunk 1 above)
Comment (SudharakaP) [reply server/graphql/v2/object/Event.js]: Yep, I thought we need it to handle the null case, but I am wrong. We don't need it. I've removed the resolver. Thanks for pointing it out. 👍🏽

File: server/graphql/v2/object/Event.js (hunk 2 above)
Comment (SudharakaP) [reply server/graphql/v2/object/Event.js:55]: Well, my thinking was that we are not using `startsAt` at this stage. Only the `endsAt` is useful for this PR (to support the warning for past events). I normally go by the rule that I try not to introduce changes which are not relevant to the issue in question which the PR aims to solve. This is why I didn't introduce it. Anyways, do you think we should introduce `startsAt`? 🤔

Comment (znarf) [reply server/graphql/v2/object/Event.js:55]: It doesn't make sense to introduce `endsAt` alone. I'm asking you to introduce `startsAt` too.

Comment (SudharakaP) [reply server/graphql/v2/object/Event.js:59]: @znarf : I've added `startsAt` as well. 👍🏽

**Review by znarf — APPROVED** [review]

**Pull request merged**